### PR TITLE
bugfix/android-iap-signal-instead-of-timer: Update the IAP project demo

### PR DESCRIPTION
**Pull request by xsellier**
Related to: https://github.com/godotengine/godot/pull/16647

### CODE REVIEW

File: misc/android_iap/iap.gd [hunk 1]
```diff
@@ -15,80 +17,92 @@ signal consume_not_required
...
+# Used to be able to stub the android behavior
+onready var is_debug_build = OS.is_debug_build() and OS.get_name() != IAP_PLATFORM
 
 func _ready():
-	if payment:
-		# set callback with this script instance
-		payment.setPurchaseCallbackId(get_instance_ID())
+  if payment:
+    # set callback with this script instance
+    payment.setPurchaseCallbackId(get_instance_ID())
```
Comment (bojidar-bg) [inline misc/android_iap/iap.gd]: Use tabs, not spaces 😃

Comment (xsellier) [reply misc/android_iap/iap.gd]: Done :)

Comment (akien-mga) [reply misc/android_iap/iap.gd]: You forgot to push I guess, it's still the same commit :)

**Comment by xsellier** [comment]
Indeed, since I amended my commit I had to force the push.

File: misc/android_iap/iap.gd [hunk 2]
```diff
@@ -27,82 +35,88 @@ func set_auto_consume(auto):
...
 		payment.setAutoConsume(auto)
 
-
 # request user owned item, callback : has_purchased
 func request_purchased():
 	if payment:
 		payment.requestPurchased()
 
 func has_purchased(receipt, signature, sku):
-	if sku == "":
```
Comment (akien-mga) [inline misc/android_iap/iap.gd]: Those changes add a lot of noise to the diff for no practical reason, and even though it's now consistent in this file, it's still inconsistent with many of the other demos where `"` is used.
So I prefer we avoid sneaking in such changes in unrelated commits, and if we decide on a given style guide for this, then we can do the change. Personally I think `"` is fine for string quotes, it's much rarer to have to use (and then escape) `"` in a string than `'`.

Comment (xsellier) [reply misc/android_iap/iap.gd]: Sure.

**Comment by aaronfranke** [comment]
What is the status of this PR? Is this fix still desired to be added to the 2.1 branch?

File: misc/android_iap/iap.gd [hunk 3]
```diff
@@ -27,7 +35,6 @@ func set_auto_consume(auto):
 	if payment:
 		payment.setAutoConsume(auto)
 
-
```
Comment (aaronfranke) [inline misc/android_iap/iap.gd]: Per the [style guide](https://docs.godotengine.org/en/latest/getting_started/scripting/gdscript/gdscript_styleguide.html), functions should be separated by two empty lines.

**Comment by xsellier** [comment]
Since no one is interested in this PR, I'm closing it.

File: misc/android_iap/iap.gd (hunk 3 above)
Comment (xsellier) [reply misc/android_iap/iap.gd]: This PR is almost 2 years old. There is no need for further review.

**Pull request closed without merging**